src/jobs/updateRole.js

```diff
@@ -0,0 +1,43 @@
+const moment = require('moment');
+const Company = require('../models/Company');
+const User = require('../models/User');
+const Role = require('../models/Role');
+const Contract = require('../models/Contract');
+const { AUXILIARY_WITHOUT_COMPANY } = require('../helpers/constants');
+const EmailHelper = require('../helpers/email');
+
+const updateRole = {
+  async method() {
+    let error;
+    let updatedUserCount = 0;
+    try {
+      const companies = await Company.find({}).lean();
+      const role = await Role.findOne({ name: AUXILIARY_WITHOUT_COMPANY }).lean();
+      for (const company of companies) {
+        const contracts = await Contract.find({
+          company: company._id,
+          endDate: moment().subtract(1, 'day').endOf('day').toDate(),
+        }).lean();
+        const usersToUpdate = contracts.map(contract => contract.user);
+        const updatedUser = await User.updateMany({ _id: { $in: usersToUpdate } }, { $set: { role: role._id } });
+        updatedUserCount += updatedUser.nModified;
+      }
```

Comment:
Tu pourrais récupérer uniquement les ids des entreprises pour les utiliser ds un `$in`. Cela éviterait d'appeler les requêtes à chaque itération:
```
const companiesId = await Company.find({}, {id: 1}).lean();
const role = await Role.findOne({ name: AUXILIARY_WITHOUT_COMPANY }).lean();
const contracts = await Contract.find({
          company: {$in: companiesId }
          endDate: moment().subtract(1, 'day').endOf('day').toDate(),
        }).lean();
const usersToUpdate = contracts.map(contract => contract.user);
const updatedUser = await User.updateMany({ _id: { $in: usersToUpdate } }, { $set: { role: role._id } });
updatedUserCount = updatedUser.nModified;
```